mods/gen2/moves.js

```diff
@@ -12,13 +12,17 @@ exports.BattleMovedex = {
 			if (target.boosts.atk >= 6) {
 				return false;
 			}
+
+			var boost = 0;
+
 			if (target.hp <= target.maxhp / 2) {
-				this.boost({atk: 2});
-				return false;
+				boost = (target.boost ? target.boost.atk : 0) + 2;
```

Comment:
The main gripe I have with that is that the investigation of the problem probably won't happen anytime soon. This bug has been in for a while and it means that belly drum is pretty much unusable in gen 2 as a result. Yes, my workaround is an ugly hack, but it does 'solve' the problem until there's something better.